### PR TITLE
Fix bug in EpochMetric, Add output_transform to ROC_AUC, AveragePrecision metrics

### LINKED ISSUE #313: Missing output_transform + bug with check once the signature
As it was remarked by @amitibo in the [comment](https://github.com/pytorch/ignite/pull/235#issuecomment-435335041) there is a output_transform missing in `__init__` of `ROC_AUC` and `AveragePrecision`. 

There is also a bug with `check once the signature` [here](https://github.com/pytorch/ignite/blob/master/ignite/metrics/epoch_metric.py#L61) if the input data is such that a single batch does not satisfy the condition required by `compute_fn` but complete data does satisfy.

### PR DESCRIPTION
**Pull request by vfdev-5**
Fixes #313 , #309 

Description:

- Added missing `output_transform` to ROC_AUC and AveragePrecision.
  - Added tests and integration tests
- Replace `raise RuntimeException` by warning in EpochMetric, as this test can potentially stop the computation: if the input data is such that a single batch does not satisfy the condition required by `compute_fn` but complete data satisfies all conditions.
- Added a warning in EpochMetric on potential memory error if input data is large

Check list:
* [X] New tests are added (if a new feature is modified)
* [X] New doc strings: text and/or example code are in RST format
* [X] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/epoch_metric.py [hunk 1]
```diff
@@ -60,7 +72,8 @@ def update(self, output):
             try:
                 self.compute_fn(self._predictions, self._targets)
             except Exception as e:
-                raise RuntimeError("Problem with `compute_fn`:\n {}".format(e))
+                warnings.warn("Probably, there can be a problem with `compute_fn`:\n {}".format(e),
```
Comment (jasonkriss) [inline ignite/metrics/epoch_metric.py]: What's the reason for switching to a warning here?

Comment (vfdev-5) [reply ignite/metrics/epoch_metric.py]: Replace raise RuntimeException by warning in EpochMetric, as this test can potentially stops the computation: if the input data is such that a single batch does not satisfy the condition required by compute_fn but complete data satisfies all conditions.

Comment (jasonkriss) [reply ignite/metrics/epoch_metric.py]: Got it. Thanks for clarifying!

**Comment by vfdev-5** [comment]
@jasonkriss please merge this if OK

**Pull request merged**